streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -49,6 +61,30 @@
         this.stateDirectory = stateDirectory;
     }
 
+    protected void initializeCheckpoint() {
+        // we will delete the local checkpoint file after registering the state stores and loading them into the
```

Comment:
Yes it is a simplification: we are going to write the checkpoint file after initialization and on the first commit we would write checkpoint file again, so that we only have a short window of risk.

Reply:
Independent of this PR, might it be worth to change this and close this window of risk?